src/main/java/dci/j24e01/OurShop/controllers/ProductController.java

```diff
@@ -49,4 +57,15 @@ public String postAdd(@ModelAttribute Product product) {
             return "redirect:/products?failure=true";
         }
     }
+
+    @GetMapping("/products/delete/{id}")
+    public String delete(@PathVariable Long id) {
```

Comment:
In the Delete category we are using Query params, here we are using a Slug.
Let's be consistent in how we use similar controllers